components/usage/pkg/apiv1/billing.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package apiv1
+
+import (
+	"context"
+	"fmt"
+	"github.com/gitpod-io/gitpod/common-go/log"
+	v1 "github.com/gitpod-io/gitpod/usage-api/v1"
+	"github.com/gitpod-io/gitpod/usage/pkg/db"
+	"github.com/gitpod-io/gitpod/usage/pkg/stripe"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"math"
+)
+
+func NewBillingService(stripeClient *stripe.Client) *BillingService {
+	return &BillingService{
+		stripeClient: stripeClient,
+	}
+}
+
+type BillingService struct {
+	stripeClient *stripe.Client
+
+	v1.UnimplementedBillingServiceServer
+}
+
+func (s *BillingService) UpdateInvoices(ctx context.Context, in *v1.UpdateInvoicesRequest) (*v1.UpdateInvoicesResponse, error) {
+	credits, err := creditSummaryForTeams(in.GetSessions())
+	if err != nil {
+		log.Log.WithError(err).Errorf("Failed to compute credit summary.")
+		return nil, status.Errorf(codes.InvalidArgument, "failed to compute credit summary")
+	}
+
+	err = s.stripeClient.UpdateUsage(credits)
+	if err != nil {
+		log.Log.WithError(err).Errorf("Failed to update stripe invoices.")
+		return nil, status.Errorf(codes.Internal, "failed to update stripe invoices")
+	}
+
+	return &v1.UpdateInvoicesResponse{}, nil
```

Comment:
Is there something useful we can return here? The number of invoices updated perhaps?

Reply:
That response is open for extension. For now, I'd keep it empty until we know what we need to return, then we can add